### PR TITLE
fix(parse): do not use `const enum`

### PR DESCRIPTION
**Pull request by peterblazejewicz**
This is now banned by linter, but barely surfaces from time to time,
like in this case:
https://github.com/DefinitelyTyped/DefinitelyTyped/runs/7225578093?check_suite_focus=true#step:6:3241

Removing as per handbook:
https://www.typescriptlang.org/docs/handbook/enums.html#const-enum-pitfalls

Thanks!

/cc @nicolas377
ref: #61115

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)

### CODE REVIEW

**Review by nicolas377 — APPROVED** [review]
Thanks for this!

**Comment by yoursunny** [comment]
A search indicates there are 173 `const enum` occurrences in this codebase.
https://github.com/DefinitelyTyped/DefinitelyTyped/search?q=%22const+enum%22&type=code

Maybe delete all the them in the same PR?

**Comment by nicolas377** [comment]
Maybe, but that PR would take significantly longer to merge, while this is a hotfix to allow PRs that modify @types/node to pass CI.

**Review by sheetalkamat — APPROVED** [review]

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
I think this should be on ParseError as statics instead of enum since I am not sure `ErrorCode` exists.

**Comment by peterblazejewicz** [comment]
@sheetalkamat you're correct:
https://github.com/parse-community/Parse-SDK-JS/blob/v1.11.0/src/ParseError.js
but that would change the focus (we would have to correct symbol definition, instead of just fixing the CI failing on v1 tests due to new linter rules). IMO, the current change is safe (removing const from `d.ts` as per handbook)

**Review by panva — APPROVED** [review]

**Comment by nicolas377** [comment]
> IMO, the current change is safe (removing const from d.ts as per handbook)

I agree with @peterblazejewicz. I'm going to turn all `const enum` instances into just `enum` instances in a separate PR (if a maintainer hasn't already), but for now, we need to unblock @types/node.

**Pull request closed without merging**
